### PR TITLE
feat(manage-aur): add capability of managing AUR repos

### PR DESCRIPTION
**Pull request by dr460nf1r3**
Adds management of AUR packages.

### CODE REVIEW

**Review by JustTNE — CHANGES_REQUESTED** [review]
I highly doubt this is compatible with github actions honestly

**Comment by dr460nf1r3** [comment]
Added CI-specific variables so both GitHub Actions/GitLab CI can be supported.

File: .ci/manage-aur.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# we are skipping debug output here to not leak the key
+set +x
+if [[ -z ${AUR_KEY+x} ]]; then
+    echo "No AUR SSH key available, backing off!"
+    exit 1
+else
+    echo "$AUR_KEY" >/app/key_aur
+    chmod 600 /app/key_aur
+fi
+set -x
```
Comment (JustTNE) [inline .ci/manage-aur.sh:16]: As discussed, let us use the existing SSH key here instead

File: .ci/manage-aur.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+# CI-specific variables
+if [ -v GITLAB_CI ]; then
+    _CI_PIPELINE_URL="$CI_PIPELINE_URL"
+    _CI_REPOSITORY_URL="$CI_PROJECT_URL"
+elif [ -v GITHUB_ACTIONS ]; then
+    _CI_PIPELINE_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}"
+    _CI_REPOSITORY_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/commits/main"
+else
+    echo "Supplying pipeline/project details in commit messages is only supported on GitLab CI/GitHub Actions."
+fi
```
Comment (JustTNE) [inline .ci/manage-aur.sh:30]:
```suggestion
# CI-specific variables
if [ -v GITLAB_CI ]; then
    _CI_PIPELINE_URL="$CI_PIPELINE_URL"
    _CI_REPOSITORY_URL="${CI_SERVER_URL}/${CI_PROJECT_PATH}/-/commits/${CI_DEFAULT_BRANCH }"
elif [ -v GITHUB_ACTIONS ]; then
    _CI_PIPELINE_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}"
    _CI_REPOSITORY_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/commits/${GITHUB_REF}"
else
    echo "Warning: Supplying pipeline/project details in commit messages is only supported on GitLab CI/GitHub Actions."
fi
```

File: .ci/manage-aur.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+# shellcheck source=/dev/null
+source .ci/util.shlib
+export GIT_SSH_COMMAND="-i /app/key_aur ssh -o StrictHostKeyChecking=accept-new"
```
Comment (JustTNE) [inline .ci/manage-aur.sh:35]: As discussed, let us use the existing SSH key here instead

File: .ci/manage-aur.sh [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    exit 0
+fi
+
+for package in "${PACKAGES[@]}"; do
+    unset VARIABLES
+    declare -A VARIABLES
+    if UTIL_READ_MANAGED_PACAKGE "$package" VARIABLES; then
+        if [[ -v "VARIABLES[CI_MANAGE_AUR]" ]] && [[ "${VARIABLES[CI_MANAGE_AUR]}" != "true" ]]; then
+            continue
+        else
```
Comment (JustTNE) [inline .ci/manage-aur.sh:57]: useless else

File: .ci/manage-aur.sh [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            echo "Starting AUR management.."
```
Comment (JustTNE) [inline .ci/manage-aur.sh:58]: should not log spam

File: .ci/manage-aur.sh [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+            # Clone via SSH to allow pushing
+            if ! test -d "$TMPDIR/aur-push/$package"; then
+                git clone "ssh://aur@aur.archlinux.org/$package.git" "$TMPDIR/aur-push/$package"
+            fi
```
Comment (JustTNE) [inline .ci/manage-aur.sh:63]:
```suggestion
            # Clone via SSH to allow pushing
            if ! git clone "ssh://aur@aur.archlinux.org/$package.git" "$TMPDIR/aur-push/$package"; then
                echo "Warning: Failed to clone \"$package\" from AUR."
                continue
            fi
```

File: .ci/manage-aur.sh [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+            # We always run shfmt on the PKGBUILD. Two runs of shfmt on the same file should not change anything
+            shfmt -w "$TMPDIR/aur-pulls/$package/PKGBUILD"
+
+            # Rsync: delete files in the destination that are not in the source. Exclude deleting .CI, exclude copying .git
```
Comment (JustTNE) [inline .ci/manage-aur.sh:68]: that kinda makes no sense

```suggestion
            # Rsync: delete files in the destination that are not in the source. Exclude copying .CI and .git
```

File: .ci/manage-aur.sh [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                git push
+            elif ! git diff --exit-code --quiet; then
+                git add .
+                git commit -m "chore: update $package" \
+                    -m "This commit was automatically generated by our CI pipeline."
+                git push
+            else
+                echo "No changes detected, skipping!"
+                continue
+            fi
```
Comment (JustTNE) [inline .ci/manage-aur.sh:88]:
```suggestion
            # Only push if there are changes
            if ! git diff --exit-code --quiet; then
                git add .
                if [ -v _CI_COMMITS_URL ]; then
                    git commit -m "chore: update $package" \
                        -m "This commit was automatically generated to reflect changes to this package in another repository." \
                        -m "The changelog for this package can be found at ${_CI_COMMITS_URL}." \
                        -m "Logs of the corresponding pipeline run can be found here: $_CI_PIPELINE_URL."
                else
                    git commit -m "chore: update $package" \
                        -m "This commit was automatically generated to reflect changes to this package in another repository." \
                        -m "Unfortunately, due to a misconfiguration, the URL of the source repository is not available."
                fi
                git push
            else
                echo "No changes detected, skipping!"
                continue
            fi
```

File: .ci/on-commit.sh [hunk 9]
```diff
@@ -115,6 +115,7 @@ else
             .ci/schedule-packages.sh schedule all
         else
             .ci/schedule-packages.sh schedule "${!PACKAGES[@]}"
+            .ci/manage-aur.sh "${!PACKAGES[@]}"
         fi
```
Comment (JustTNE) [inline .ci/on-commit.sh:119]:
```suggestion
            .ci/schedule-packages.sh schedule all
            .ci/manage-aur.sh all
        else
            .ci/schedule-packages.sh schedule "${!PACKAGES[@]}"
            .ci/manage-aur.sh "${!PACKAGES[@]}"
        fi
```

File: .github/workflows/chaotic-on-commit.yml [hunk 10]
```diff
@@ -6,6 +6,7 @@ permissions:
 
 env:
   ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN || secrets.GITHUB_TOKEN }}
+  ACCESS_TOKEN: ${{ secrets.AUR_KEY || secrets.AUR_KEY }}
```
Comment (JustTNE) [inline .github/workflows/chaotic-on-commit.yml:9]: ...what?

File: .github/workflows/chaotic-on-commit.yml [hunk 11]
```diff
@@ -52,5 +53,6 @@ jobs:
           SCHEDULE_PARAMS: ${{ needs.process.outputs.SCHEDULE_PARAMS }}
           REDIS_PASSWORD: ${{ secrets.REDIS_PASSWORD }}
           DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+          AUR_KEY: ${{ secrets.AUR_KEY }}
```
Comment (JustTNE) [inline .github/workflows/chaotic-on-commit.yml:56]: uhm..

**Review by JustTNE — CHANGES_REQUESTED** [review]

**Pull request merged**